backends/dpdk/dpdkArch.h

```diff
@@ -894,26 +902,59 @@ class SplitActionSelectorTable : public SplitP4TableCommon {
  */
 class SplitActionProfileTable : public SplitP4TableCommon {
  public:
-    SplitActionProfileTable(P4::ReferenceMap* refMap, P4::TypeMap* typeMap,
-                            DpdkProgramStructure* structure)
-        : SplitP4TableCommon(refMap, typeMap, structure) {
+    SplitActionProfileTable(
+        P4::ReferenceMap* refMap, P4::TypeMap* typeMap, DpdkProgramStructure* structure,
+        std::map<cstring, std::vector<std::tuple<cstring, IR::Constant*>>>& actionCaseMap)
+        : SplitP4TableCommon(refMap, typeMap, structure, actionCaseMap) {
         implementation = TableImplementation::ACTION_PROFILE;
     }
     const IR::Node* postorder(IR::P4Table* tbl) override;
 };
 
+class UpdateActionForSwitch : public Transform {
+    std::map<cstring, std::vector<std::tuple<cstring, IR::Constant*>>>& actionCaseMap;
+
+ public:
+    explicit UpdateActionForSwitch(
+        std::map<cstring, std::vector<std::tuple<cstring, IR::Constant*>>>& actionCaseMap)
+        : actionCaseMap(actionCaseMap) {
+        setName("UpdateActionForSwitch");
+    }
+    const IR::Node* postorder(IR::P4Action* action) {
+        if (actionCaseMap.count(action->name.name)) {
+            auto acm = actionCaseMap[action->name.name];
+            auto body = new IR::BlockStatement(action->body->srcInfo);
+            for (auto pair : acm) {
+                auto assn = new IR::AssignmentStatement(
+                    new IR::PathExpression(IR::ID(get<0>(pair))), get<1>(pair));
+                body->push_back(assn);
+            }
+            for (auto s : action->body->components) body->push_back(s);
+            action->body = body;
+            actionCaseMap.erase(action->name.name);
+        }
+        return action;
+    }
+};
+
 /**
- * Handle ActionSelector and ActionProfile extern in PSA
- */
+ * Handle ActionSelector and ActionProfile extern */
 class ConvertActionSelectorAndProfile : public PassManager {
+    // Map which holds the switch expression variable and constant tuple per switch statement for
```

Comment:
I see the documentation here.
But if you define a separate class, that's the natural place to put it.